xarray/plot/plot.py

```diff
@@ -404,12 +428,27 @@ def line(
     *args, **kwargs : optional
         Additional arguments to :py:func:`matplotlib:matplotlib.pyplot.plot`.
     """
+
+    from ..core.groupby import GroupBy
+
+    if isinstance(darray, GroupBy):
+        if row is None and col is None:
+            raise ValueError(
+                f"Both 'row' and 'col' are None. Expected one of 'row' or 'col' to be '{darray._unique_coord.name}'"
```

Comment:
Not a fan of this error. I was expecting `ds.variable.groupby(ds.id).plot.line()` to work.

If no facetgrid options have been defined it should just loop through the plots and append to the current plot with  different colors, for example:
```python
ds = xr.Dataset(
            {
                "variable": (
                    ("lat", "lon", "time"),
                    np.arange(72.0).reshape((4, 3, 6)),
                ),
                "id": (("lat", "lon"), np.arange(12.0).reshape((4, 3))),
            },
            coords={
                "lat": np.arange(4),
                "lon": np.arange(3),
                "time": pd.date_range(start="2001-01-01", freq="12H", periods=6),
            },
        )

for id_val, var in ds.variable.groupby(ds.id):
    var.plot.line(x="time")
```
![image](https://user-images.githubusercontent.com/14371165/129421565-22b5e1be-68b1-495d-9e50-b5cdf9946447.png)

Compared to facetgrid version:
```python
ds.variable.groupby(ds.id).plot(row="id")
```
![image](https://user-images.githubusercontent.com/14371165/129422086-e642d781-f760-458d-abd4-f9725081be0e.png)

Not sure how difficult this is to implement but this can also be saved for a future PR.